qiskit/chemistry/components/variational_forms/uvcc.py

```diff
@@ -0,0 +1,284 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+""" Unitary Vibrational Coupled-Cluster Single and Double excitations variational form. """
+
+import logging
+import sys
+from typing import Optional, List, Tuple, Union, cast
+
+import numpy as np
+
+from qiskit import QuantumRegister, QuantumCircuit
+from qiskit.tools import parallel_map
+from qiskit.tools.events import TextProgressBar
+
+from qiskit.circuit import ParameterVector, Parameter
+
+from qiskit.aqua import aqua_globals
+from qiskit.aqua.operators import WeightedPauliOperator
+from qiskit.aqua.components.initial_states import InitialState
+from qiskit.aqua.components.variational_forms import VariationalForm
+from qiskit.chemistry.bosonic_operator import BosonicOperator
+
+logger = logging.getLogger(__name__)
+
+
+class UVCC(VariationalForm):
+    """
+    This trial wavefunction is a Unitary Vibrational Coupled-Cluster Single and Double excitations
+    variational form.
+    For more information, see Ollitrault Pauline J., Chemical science 11 (2020): 6842-6855.
+    """
+
+    def __init__(self, num_qubits: int,
```

Comment:
Same comment as CHC about `num_qubits`

Reply:
Updated in da221021bab30929d64f160b5db54d83f087c918